journal.py

```diff
@@ -11,107 +10,136 @@
+
+    def json_detail(self):
+        '''Displays a JSON object for detaile view'''
+        return {'title': self.title,
+                'text': self.render_markdown(),
+                'created': self.created.strftime('%b %d, %Y'),
+                'id': self.id}
+
+    def json_edit(self):
+        '''Does not show HTML during editing'''
+        return {'title': self.title,
+                'text': self.text,
+                'created': self.created.strftime('%b %d, %Y'),
+                'id': self.id}
+
+    def editing(self, request):
+        '''Allows for updating both the entry text, as well as the title'''
+        self.title = request.params.get('title', None)
+        self.text = request.params.get('text', None)
+
+    def render_markdown(self):
+        '''Used to put markdown on text within my template'''
+        return markdown.markdown(
+            self.text, extensions=['codehilite', 'fenced_code'])
 
-NEW_ENTRY = """
-SELECT * FROM entries ORDER BY created DESC LIMIT 1
-"""
 
 logging.basicConfig()
 log = logging.getLogger(__file__)
 
 
-def connect_db(settings):
-    """Return a connection to the configured database"""
-    return psycopg2.connect(settings['db'])
-
-
-def init_db():
-    """Create database dables defined by DB_SCHEMA
-
-    Warning: This function will not update existing table definitions
-    """
-    settings = {}
-    settings['db'] = os.environ.get(
-        'DATABASE_URL', 'dbname=learning_journal user=nbeck'
-    )
-    with closing(connect_db(settings)) as db:
-        db.cursor().execute(DB_SCHEMA)
-        db.commit()
-
-
-@subscriber(NewRequest)
-def open_connection(event):
-    request = event.request
-    settings = request.registry.settings
-    request.db = connect_db(settings)
-    request.add_finished_callback(close_connection)
-
-
-def close_connection(request):
-    """close the database connection for this request
-
-    If there has been an error in the processing of the request, abort any
-    open transactions.
-    """
-    db = getattr(request, 'db', None)
-    if db is not None:
-        if request.exception is not None:
-            db.rollback()
-        else:
-            db.commit()
-        request.db.close()
-
-
 def main():
     """Create a configured wsgi app"""
     settings = {}
     settings['reload_all'] = os.environ.get('DEBUG', True)
     settings['debug_all'] = os.environ.get('DEBUG', True)
-    settings['db'] = os.environ.get(
-        'DATABASE_URL', 'dbname=learning_journal user=nbeck'
+    settings['sqlalchemy.url'] = os.environ.get(
+        'DATABASE_URL', 'postgresql://nbeck:@localhost:5432/learning_journal'
     )
-    # secret value for session signing:
-    settings['auth.username'] = os.environ.get('ADMIN_USERNAME', 'admin')
+    engine = sa.engine_from_config(settings, 'sqlalchemy.')
+    DBSession.configure(bind=engine)
+    settings['auth.username'] = os.environ.get('AUTH_USERNAME', 'admin')
```

Comment:
Don't forget to make sure that you update the environmental variable on Heroku before you deploy these changes.